apps/frontend/src/components/fap/Proposals/FapProposalsAndAssignmentsTable.tsx

```diff
@@ -464,10 +468,12 @@ const FapProposalsAndAssignmentsTable = ({
             assignedReviewer.fapMemberUserId === data.fapChair?.id
               ? data.fapChairProposalCount! - 1
               : data.fapChairProposalCount,
-          fapSecretaryProposalCount:
-            assignedReviewer.fapMemberUserId === data.fapSecretary?.id
-              ? data.fapSecretaryProposalCount! - 1
-              : data.fapSecretaryProposalCount,
+          fapSecretaryProposalCount: data.fapSecretaryProposalCount.map(
+            (value, index) =>
+              assignedReviewer.fapMemberUserId === data.fapSecretary[index].id
+                ? value - 1
+                : value
+          ),
```

Comment:
I don't understand this code. When I try to find the origins of `data.fapSecretaryProposalCount` I ended up here: https://github.com/UserOfficeProject/user-office-core/blob/develop/apps/backend/src/datasources/postgres/FapDataSource.ts#L319 which seems like it should just return the number, but then we do some mapping and id checks and conditional subtractions. May be lack of familiarity with the framework, but I'm feeling clueless, can you explain it to me?

Reply:
I think this is required as the Secretary and Chairs details, including Proposal Count, are returned as part of the `getFap` call that is done when opening the page to edit the Panel. We don't refetch the FAP when we go between the tabs, but we do refetch the reviewers when we load the `Members` page which updates the review counts for all the reviewers but not the Secretaries and chairs. So this keeps track of the reviews for the Secretaries and Chairs count tho I agree it is a be overly complex and unclear